firefox-ios/firefox-ios-tests/Tests/ClientTests/Coordinators/LaunchView/LaunchScreenViewModelTests.swift

```diff
@@ -135,7 +122,375 @@ final class LaunchScreenViewModelTests: XCTestCase {
+        profile.prefs.setInt(1, forKey: PrefsKeys.IntroSeen)
+        profile.prefs.setString("112.0", forKey: PrefsKeys.AppVersion.Latest)
+        profile.hasSyncableAccountMock = true
+        let message = createMessage()
+        messageManager.message = message
+
+        let subject = createSubject()
+        subject.delegate = delegate
+        subject.startLoading(appVersion: "113.0")
+
+        XCTAssertEqual(subject.launchOrder.count, 1)
+        subject.loadNextLaunchType()
+        assertSavedLaunchType(.update)
+    }
+
+    // MARK: - Splash Screen Experiment Tests
+
+    func testSplashScreenExperiment_initiallyNotShown() {
+        let subject = createSubject()
+        XCTAssertFalse(subject.getSplashScreenExperimentHasShown())
+    }
+
+    func testSplashScreenExperiment_setThenGet_returnsTrue() {
+        let subject = createSubject()
+        subject.setSplashScreenExperimentHasShown()
+        XCTAssertTrue(subject.getSplashScreenExperimentHasShown())
+    }
+
+    func testSplashScreenExperiment_multipleSets_staysTrue() {
+        let subject = createSubject()
+        subject.setSplashScreenExperimentHasShown()
+        subject.setSplashScreenExperimentHasShown()
+        subject.setSplashScreenExperimentHasShown()
+        XCTAssertTrue(subject.getSplashScreenExperimentHasShown())
+    }
+
     // MARK: - Helpers
+
+    private func setTermsOfServiceFeatureEnabled(_ enabled: Bool) {
+        FxNimbus.shared.features.tosFeature.with(initializer: { _, _ in
+            TosFeature(status: enabled)
+        })
+    }
+
+    private func assertSavedLaunchType(
+        _ expectedCase: LaunchTypeCase,
+        file: StaticString = #filePath,
+        line: UInt = #line
+    ) {
+        guard let saved = delegate.savedLaunchType else {
+            XCTFail("Expected \(expectedCase.description) but savedLaunchType was nil", file: file, line: line)
+            return
+        }
+
+        let expected: LaunchType
+        switch expectedCase {
+        case .intro:
+            expected = .intro(manager: IntroScreenManager(prefs: profile.prefs))
+        case .termsOfService:
+            expected = .termsOfService(manager: TermsOfServiceManager(prefs: profile.prefs))
+        case .update:
+            let onboardingModel = createOnboardingViewModel()
+            let telemetryUtility = OnboardingTelemetryUtility(with: onboardingModel)
+            expected = .update(viewModel: UpdateViewModel(profile: profile,
+                                                          model: onboardingModel,
+                                                          telemetryUtility: telemetryUtility,
+                                                          windowUUID: windowUUID))
+        case .survey:
+            expected = .survey(manager: SurveySurfaceManager(windowUUID: windowUUID, and: messageManager))
+        case .defaultBrowser:
+            expected = .defaultBrowser
+        }
```

Comment:
instead of adding this here maybe we can pass in the expected case instead? so we can see the interface and we can keep this method small and generic?

Curious why we need to add these switch checks, where we were didn't need it before 
```
guard case .intro = delegate.savedLaunchType else {
            XCTFail("Expected intro, but was \(String(describing: delegate.savedLaunchType))")
            return
        }
```
before